### ISSUE TITLE
.NET Native Content pipeline issue

### ISSUE THREAD

**Issue by jackdolabany** [MonoGame/MonoGame#4868]
I've been having issues for months now getting the content to build in my Windows 10 Universal project using the .NET Native toolchain. I recently started a thread on the forums here:

http://community.monogame.net/t/custom-content-pipeline-serialization-issues-in-windows-10/7665/3

I finally tracked the issue down to this line of code: https://github.com/mono/MonoGame/blob/develop/MonoGame.Framework/Content/ContentExtensions.cs#L37

It looks like it's trying to filter out properties on the object that have been hidden using the "new" keyword in C#. Unfortunately, the trick of checking if the GetMethods for the properties are reference equal doesn't seem to work using the .NET Native toolchain. In general, this doesn’t feel like a great way to make this check. The result is that most custom content loads up with blank or null values, which causes crashes elsewhere. 

For my game, I just commented out that line and everything works fine. I don't do any property hiding, although I'm not sure if things would break outright even if I did. Regardless, I spent some time investigating fixes to the problem and it turns out .NET doesn't really have a great way to determine if one property is hiding another in a base class. Especially in the RT environment where reflection is limited. If possible, it seems like it might be best to not do this check at all. I’m not sure what, if anything, was actually breaking downstream before this check existed. Like I said, my project works fine without it. 

To find out why this code exists in the first place, I tried to track down the original commit, but haven't had much luck. I think there have been some git history rewrite shenanigans going on. You can see that in this commit (https://github.com/mono/MonoGame/commit/560741319e82c396049def24e2fdaa32d39e58cf) the logic was moved from ReflectiveReader.cs to ContentExtensions.cs. 

But if you go through the history of ReflectiveReader.cs, you won't see that commit. Also, I can't find any trace of the property hiding check ever having been in that file. Instead you can see the changed seems to have happened between this commit (https://github.com/mono/MonoGame/commit/b940c0d9a289f191e08a17137c06290ebb4a18a2) and this commit (https://github.com/mono/MonoGame/commit/c99103b5e435a15fa24b0ea89f9bf4b2261f35a9)

TLDR; 
The check for property hiding is sketchball and breaks .NET Native apps. .NET doesn’t provide a great way to check for this at all. Maybe you should kill it altogether. 

PS: The GetAllProperties() and GetAllFields() functions are misnamed. They both get very restrictive sets of properties/fields.

**Comment by MattWhilden** [comment]
I work on the .NET Native runtime and compiler team. Do you know which version of VS/UWP tools you are using? We did work around these areas for Update 2 and 3 so maybe giving those a shot can help clear things up. If it's still reproing with new tools, please let us know. Happy to help work around issues and get things patched up in the compiler. This kind of behavior is definitely not "by design". :-)

**Comment by jackdolabany** [comment]
Still happening. I'm using VS 2015. Visual Studio Tools for Universal Windows Apps 14.0.25123.00. The app is referencing the dll with this Identity: UAP,Version=10.0.10586.0

**Comment by jackdolabany** [comment]
Here's a small app isolating the problem

https://github.com/jackdolabany/TestForDotNetBug

**Comment by MattWhilden** [comment]
Thanks @jackdolabany! I'll track someone down to take a look. I'd expect someone to triage this today or tomorrow at the latest. Feel free to chastise me if we become delinquent... easy to become distracted. :-)

**Comment by MichalStrehovsky** [comment]
The right solution here might be to update MonoGame to call the Equals override.

.NET Core profile [doesn't provide](https://msdn.microsoft.com/en-us/library/system.reflection.methodinfo.op_equality(v=vs.110).aspx) an overload for the `==` operator and therefore MonoGame is getting the default (referential equality) comparison. The existing code works on CoreCLR by luck because CoreCLR's implementation of reflection does more aggressive caching which leads to reuse of existing objects.

**Comment by jackdolabany** [comment]
That didn't work when I tried it back in May (when I first opened this issue). I'll try again. The reference equals seems hacky anyway. If it works now I'll make a PR for it and we'll be good to go.

**Comment by MattWhilden** [comment]
Perfect. Let us know if there's anything else weird.

**Comment by harry-cpp** [comment]
There is corert now, and MonoGame works with it just fine, so closing this.
